src/feat/mel-computations.cc

```diff
@@ -129,29 +108,93 @@ MelBanks::MelBanks(const MelBanksOptions &opts,
     bins_[bin].second.CopyFromVec(this_bin.Range(first_index, size));
 
     // Replicate a bug in HTK, for testing purposes.
-    if (opts.htk_mode && bin == 0 && mel_low_freq != 0.0)
+    if (htk_mode && bin == 0 && low_freq_ != 0.0)
       bins_[bin].second(0) = 0.0;
-
   }
-  if (debug_) {
-    for (size_t i = 0; i < bins_.size(); i++) {
-      KALDI_LOG << "bin " << i << ", offset = " << bins_[i].first
-                << ", vec = " << bins_[i].second;
+}
+
+/*
+  Notes on the shape of the modified bins.
+
+  They are shaped like a cosine function from -pi/2 to pi/2 (unlike the standard
+  triangular bins).  We define their diameter as the distance between the
+  first and last nonzero value (pi for the canonical function).  We choose
+  the diameter as:
+       d = sqrt(d1^2 + d2^2)
+  (this function may be viewed as a kind of soft-max), where d1 and d2 are
+  two different formulas for the diameter that we describe below.
+
+    d1 is a formula that ensures the bins overlap by at least a minimal amount.
+
+   Let bin_diff be the difference in Hz between this bin's center-frequency
+   and the next bin's center-frequency, or (if this is the last bin),
+   the user-specified `high-freq` which is the top of the range of frequencies
+   we cover.  Then:
+
+       d1 = 1.1 * bin_diff
+
+   The formula for d2 is designed to provide a reasonable floor so the bandwidth
+   don't get ridiculously narrow as we add more bins, and to approximate what we
+   observed the filter diameters to look like when learning filterbanks via DNNs.
+   The formula is:
+
+       d2 = 50 + 50 * f / (f + 700)
+
+   which roughly means: start with a diameter of 50Hz, increasing gradually to
+   100Hz for bins with center frequency more than about 700Hz.  There is no
+   rocket science behind this formula; it was obtained through a combination of
+   trying to match the DNN-learned filterbank bandwidths (cite: Pegah's thesis),
+   and manual tuning.
+ */
+void MelBanks::ComputeModifiedBins() {
+  int32 num_bins = center_freqs_.Dim();
+  for (int32 bin = 0; bin < num_bins; bin++) {
+    BaseFloat center_freq = center_freqs_(bin),
+        next_center = (bin == num_bins - 1 ?
+                       high_freq_ : center_freqs_(bin + 1));
+
+    BaseFloat d1 = (next_center - center_freq) * 1.1,
+              d2 = 60.0 + 50.0 * (center_freq / (center_freq + 700.0));
```

Comment:
It should be breakpoint_ instead of 700.

Reply:
I was thinking of making this a separate issue from the breakpoint, so that when we tune it it doesn't have to have the same value. Can you compare 700 with 300 (our current breakpoitn) in your new experiments?  If 300 works better we can reconsider.